lib/Doctrine/DBAL/Types/DateTimeType.php

```diff
@@ -82,4 +82,16 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
 
         return $val;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+     public function handleTypeConversion($value, AbstractPlatform $platform)
```

Comment:
When I suggested adding a new method, I was thinking of something like `isValueOfType($value)`. So that the calling code could check it first and then call the conversion function if needed.

If in 3.0 we get rid of type inheritance, every `Type` interface implementor would have to have the same copy/pasted implementation of this method.